rasa/core/channels/channel.py

```diff
@@ -81,7 +81,9 @@ def __init__(
 def register(
     input_channels: List["InputChannel"], app: Sanic, route: Optional[Text]
 ) -> None:
-    async def handler(*args, **kwargs):
+    """Registers input channel blueprints with the application."""
+
+    async def handler(*args: Any, **kwargs: Any) -> None:
         await app.agent.handle_message(*args, **kwargs)
```

Comment:
this probably doesn't cause a mypy error because we have suppressed this kind of error. However, according to this [line](https://github.com/RasaHQ/rasa/blob/82e98163d3701f490716760213b92619283a4a00/rasa/core/channels/channel.py#L111) the handler should be of type:

```suggestion
    async def handler(message: UserMessage) -> None:
        await app.agent.handle_message(message)
```

Reply:
Hmm, doesn't that line 111 actually show the output of `handler` function is a UserMessage that becomes the input for `channel.blueprint()'? When I try to find the declaration of `agent.handle_message` it appears with `*args` and `**kwargs` [here](https://github.com/RasaHQ/rasa/blob/7e9db8cd6c97b8109a2cd138b16ab635815d7cc6/rasa/core/agent.py#L563) ... unless I'm looking in the wrong place? 🤔

Reply:
The output of `handler` is `None`. The `handler` itself (a function) becomes input to `channel.blueprint`.

You're absolutely right about the `*args`, `**kwargs` thing 💯 The interface definition [here](https://github.com/RasaHQ/rasa/blob/82e98163d3701f490716760213b92619283a4a00/rasa/core/channels/channel.py#L111) specifies the handler as a function which **only**  receives a `UserMessage` object and nothing else. In my opinion these two (interface definition and `handler` need to be consistent`. I prefer the interface definition as this is also the way how the handler (aka `on_new_message`) is used by the channels

Reply:
I see now, thanks for the explanation 👍